packages/lib-classifier/src/store/SubjectStore.spec.js

```diff
@@ -77,7 +89,7 @@ describe('Model > SubjectStore', function () {
         })
 
         it('should request more subjects', function () {
-          while (rootStore.subjects.resources.size > MINIMUM_QUEUE_SIZE) {
+          while (rootStore.subjects.resources.size >= MINIMUM_QUEUE_SIZE) {
```

Comment:
This should break the tests by triggering two requests to refill the queue, instead of one. One request from inside this while loop, then another when the queue advances on the next line. 

The while loop here should reduce the queue down to the max size, if it’s larger, or do nothing if the queue is already at or below the limit. The next line is then guaranteed to advance the queue and trigger a refill.

Reply:
If I run the subject queue tests in isolation, they fail because of the extra advance call added here:
```
14 passing (1s)
  1 failing

  1) Model > SubjectStore
       Actions > advance
         with less than three subjects in the queue
           when the initial response has ten subjects
             should request more subjects:
     expected populateQueue to have been called exactly twice, but it was called thrice
    populateQueue() => [Promise] {  } at populateQueue (/usr/src/packages/lib-classifier/src/store/SubjectStore.js:55:16)
    populateQueue() => [Promise] {  } at Object.populateQueue (/usr/src/packages/lib-classifier/src/store/SubjectStore.js:110:14)
    populateQueue() => [Promise] {  } at Object.populateQueue (/usr/src/packages/lib-classifier/src/store/SubjectStore.js:110:14)
  AssertionError: expected populateQueue to have been called exactly twice, but it was called thrice
      populateQueue() => [Promise] {  } at populateQueue (src/store/SubjectStore.js:55:16)
      populateQueue() => [Promise] {  } at Object.populateQueue (src/store/SubjectStore.js:110:14)
      populateQueue() => [Promise] {  } at Object.populateQueue (src/store/SubjectStore.js:110:14)
      at Context.calledTwice (src/store/SubjectStore.spec.js:97:65)
```

Reply:
Logging the queue size inside the while loop gives different results if the tests are run in isolation or as part of the full test suite:

With `.only()` advancing the queue reduces the size by one, as expected:
```
9
8
7
6
5
4
3
Fetching more subjects
2
Fetching more subjects
          1) should request more subjects
```

Without `.only()` the queue has to be advanced twice before the subject changes:
```
9
9
8
8
7
7
6
6
5
5
4
4
3
3
Fetching more subjects
2
```

Reply:
On further investigation, the workflow store tests define `rootStore` as a global variable, which is then interfering with the store state in other tests. Removing that global will probably fix the subject store tests.
https://github.com/zooniverse/front-end-monorepo/blob/3e09df2017e02b0ee1359c2a73e65ed2513fe442/packages/lib-classifier/src/store/WorkflowStore.spec.js#L10-L17

Reply:
Good catch. I'll take a closer look.

Reply:
I spent a couple of hours looking at this today but didn’t get very far. All I figured was that tests fail if you call `advance` more than once, and then only if run in conjunction with the other store tests. 

The workaround seems to be only call advance once after setting up the mocks.

Reply:
#1083 fixes one set of breaking tests for the subject store. I've tried removing/renaming `rootStore` and `setupStores` in the PR, in case they were conflicting with other stores, but advancing the queue still fails if it's called more than once.

Reply:
I think I've found the bug. If I remove the `process.browser` check from the feedback middleware, then here's what this test logs.
```
ADVANCING
9
Aborting subject advance and showing feedback
9
ADVANCING
8
Aborting subject advance and showing feedback
8
ADVANCING
7
Aborting subject advance and showing feedback
7
ADVANCING
6
Aborting subject advance and showing feedback
6
ADVANCING
5
Aborting subject advance and showing feedback
5
ADVANCING
4
Aborting subject advance and showing feedback
4
ADVANCING
3
finished loop
Aborting subject advance and showing feedback
--------------
```

Reply:
The feedback store is aborting the subject advance action on every other call.

Reply:
Feedback is active and has 2 rules set in the mock store. I'm going to guess that the Factory classes are singletons, and that an earlier test enabled feedback on the mock project and workflow?